packages/open-next/src/core/routing/cacheInterceptor.ts

```diff
@@ -0,0 +1,202 @@
+import { createHash } from "node:crypto";
+
+import { NextConfig, PrerenderManifest } from "config/index";
+import { InternalEvent, InternalResult } from "types/open-next";
+import { emptyReadableStream, toReadableStream } from "utils/stream";
+
+import { debug } from "../../adapters/logger";
+import { CacheValue } from "../../cache/incremental/types";
+import { localizePath } from "./i18n";
+import { generateMessageGroupId } from "./util";
+
+const CACHE_ONE_YEAR = 60 * 60 * 24 * 365;
+const CACHE_ONE_MONTH = 60 * 60 * 24 * 30;
+
+async function computeCacheControl(
+  path: string,
+  body: string,
+  host: string,
+  revalidate?: number | false,
+  lastModified?: number,
+) {
+  let finalRevalidate = CACHE_ONE_YEAR;
+
+  const existingRoute = Object.entries(PrerenderManifest.routes).find(
+    (p) => p[0] === path,
+  )?.[1];
+  if (revalidate === undefined && existingRoute) {
+    finalRevalidate =
+      existingRoute.initialRevalidateSeconds === false
+        ? CACHE_ONE_YEAR
+        : existingRoute.initialRevalidateSeconds;
+    // eslint-disable-next-line sonarjs/elseif-without-else
+  } else if (revalidate !== undefined) {
+    finalRevalidate = revalidate === false ? CACHE_ONE_YEAR : revalidate;
+  }
+  // calculate age
+  const age = Math.round((Date.now() - (lastModified ?? 0)) / 1000);
+  const hash = (str: string) => createHash("md5").update(str).digest("hex");
+  const etag = hash(body);
+  if (revalidate === 0) {
+    // This one should never happen
+    return {
+      "cache-control":
+        "private, no-cache, no-store, max-age=0, must-revalidate",
+      "x-opennext-cache": "ERROR",
+      etag,
+    };
+  } else if (finalRevalidate !== CACHE_ONE_YEAR) {
+    const sMaxAge = Math.max(finalRevalidate - age, 1);
+    debug("sMaxAge", {
+      finalRevalidate,
+      age,
+      lastModified,
+      revalidate,
+    });
+    const isStale = sMaxAge === 1;
+    if (isStale) {
+      const url = NextConfig.trailingSlash ? `${path}/` : path;
```

Comment:
Would this handle the case of `basePath`, or is that irrelevant here?

Reply:
Good call, this probably won't work with `basePath`. 
If this options is set i'm pretty sure we need to remove the basePath before doing anything here. ( I need to verify this, i might be wrong )
To be fair it will be pretty much invisible for the user, it will just fail to find anything here, and it will forward the request to NextServer which should find it as expected (even using the cache)

Reply:
I'll finish this tomorrow morning with the fix for the `basePath`. Hopefully the fix is not too hard and everything works in e2e :crossed_fingers: